internal/router/router.go

```diff
@@ -78,14 +78,14 @@ func ConfiguredRouter(
 
 	router.Post("/rpc", func(rw http.ResponseWriter, r *http.Request) {
 		server.ServeHTTP(rw, r)
-	}, middlewares.Authentication, rateLim, middlewares.OtelHTTP("rpc"))
+	}, middlewares.WithLogging, middlewares.Authentication, rateLim, middlewares.OtelHTTP("rpc"))
```

Comment:
This is something that tricked me yesterday: the `OtelHTTP` middleware is after the `rateLim` middleware. That means that the requests/s we see in Grafana can be misleading. Those numbers are after rate limiting. That means we could be getting a lot more requests. 

I did a stress test locally and staging and could not get the same number of goroutines and heap memory. That's when I realized the server was being hit with way more requests and this [chart](https://console.cloud.google.com/monitoring/metrics-explorer?pageState=%7B%22xyChart%22:%7B%22dataSets%22:%5B%7B%22timeSeriesQuery%22:%22def%20fill_missing_with_default%20default_value,%20fill_period%20%3D%5Cn%20%20%7B%20t_current_time_only:%5Cn%20%20%20%20%20%20ident%5Cn%20%20%20%20%20%20%7C%20every%201m%5Cn%20%20;%20t_placeholder_through_fill_period_up_to_any_stream_present_point:%5Cn%20%20%20%20%20%20%7B%20ident%20%7C%20align%20next_older($fill_period);%20ident%20%7C%20group_by%20%5B%5D%20%7D%5Cn%20%20%20%20%20%20%7C%20join%5Cn%20%20%20%20%20%20%7C%20value%20val(1)%5Cn%20%20%20%20%20%20%7C%20align%20next_younger($fill_period)%5Cn%20%20%20%20%20%20%7C%20every%201m%20%7D%5Cn%20%20%7C%20outer_join%20%5B$default_value%5D%5Cn%20%20%7C%20value%20%5Bvalue_with_defaults:%20val(0)%5D;%5Cn%5Cndef%5Cnformat_remote_type(%5Cn%20%20remote_location_type,%20local_zone,%20local_region,%20remote_zone,%20remote_region)%5Cn%3D%5Cnif($remote_location_type%20%3D%3D%20%27EXTERNAL%27,%5Cn%20%20%27External%20(or%20VMs%20Different%20Project)%27,%5Cn%20%20if($remote_location_type%20%3D%3D%20%27GOOGLE_INTERNAL%27,%20%27Google%20Services%27,%5Cn%20%20%20%20if($remote_location_type%20%3D%3D%20%27ON_PREM%27,%20%27On-Prem%20(Interconnect%2FVPN)%27,%5Cn%20%20%20%20%20%20if($remote_location_type%20!%3D%20%27CLOUD%27,%20$remote_location_type,%5Cn%20%20%20%20%20%20%20%20if($remote_zone%20%3D%3D%20%27REMOTE_IS_EXTERNAL%27,%5Cn%20%20%20%20%20%20%20%20%20%20%27VMs%20in%20Different%20Project%27,%5Cn%20%20%20%20%20%20%20%20%20%20if($remote_zone%20%3D%3D%20$local_zone,%5Cn%20%20%20%20%20%20%20%20%20%20%20%20concatenate(concatenate(%27VMs%20Same%20Zone%20(in%20%27,%20$local_zone),%20%27)%27),%5Cn%20%20%20%20%20%20%20%20%20%20%20%20if($remote_region%20%3D%3D%20$local_region,%5Cn%20%20%20%20%20%20%20%20%20%20%20%20%20%20concatenate(concatenate(%27VMs%20Different%20Zone%20(in%20%27,%20$local_region),%20%27)%27),%5Cn%20%20%20%20%20%20%20%20%20%20%20%20%20%20concatenate(concatenate(%27VMs%20Different%20Region%20(outside%20%27,%20$local_region),%20%27)%27),)))))));%5Cn%5Cndef%20vm_flow_by_remote_type%20%3D%5Cn%20%20align%20rate(1m)%5Cn%20%20%7C%20group_by%20%5Bmetric.remote_location_type,%20metric.remote_zone,%20resource.zone%5D,%201m%5Cn%20%20%7C%20map%5Cn%20%20%20%20%20%20add%5B%5Cn%20%20%20%20%20%20%20%20local_region:%20re_extract(resource.zone,%20%27(%5B%5E-%5D%2B-%5B%5E-%5D%2B)-%5B%5E-%5D%2B%27,%20%27%5C%5C%5C%5C1%27),%5Cn%20%20%20%20%20%20%20%20remote_region:%5Cn%20%20%20%20%20%20%20%20%20%20re_extract(metric.remote_zone,%20%27(%5B%5E-%5D%2B-%5B%5E-%5D%2B)-%5B%5E-%5D%2B%27,%20%27%5C%5C%5C%5C1%27)%5D%5Cn%20%20%7C%20map%5Cn%20%20%20%20%20%20add%5B%5Cn%20%20%20%20%20%20%20%20remote_type:%5Cn%20%20%20%20%20%20%20%20%20%20@format_remote_type(metric.remote_location_type,%20resource.zone,%5Cn%20%20%20%20%20%20%20%20%20%20%20%20local_region,%20remote_zone,%20remote_region)%5D%5Cn%20%20%7C%20group_by%20%5Bremote_type%5D,%201m;%5Cn%5Cnfetch%20gce_instance%5Cn%7C%20filter%5Cn%20%20%20%20resource.instance_id%20%3D%3D%20%275246408709637053760%27%5Cn%20%20%20%20%26%26%20resource.project_id%20%3D%3D%20%27textile-310716%27%5Cn%20%20%20%20%26%26%20resource.zone%20%3D%3D%20%27us-west1-b%27%5Cn%7C%20metric%20%27networking.googleapis.com%2Fvm_flow%2Fconnection_count%27%5Cn%7C%20@vm_flow_by_remote_type%5Cn%7C%20@fill_missing_with_default%200.0%20%271%2Fs%27,%201h%22,%22targetAxis%22:%22Y1%22,%22plotType%22:%22STACKED_AREA%22%7D%5D,%22options%22:%7B%22mode%22:%22COLOR%22%7D,%22constantLines%22:%5B%5D,%22timeshiftDuration%22:%220s%22,%22y1Axis%22:%7B%22label%22:%22y1Axis%22,%22scale%22:%22LINEAR%22%7D%7D,%22isAutoRefresh%22:true,%22timeSelection%22:%7B%22timeRange%22:%221h%22%7D%7D&project=textile-310716) comproved that, and things made sense.

I wonder if we should have an HTTP middleware before `rateLim`. Too many middleware haha?

Reply:
Yeah, I've been always wondering the same but I don't recall we did it for an intentional reason.

I'd say we should put it even before `rateLim`, possible before `Authentication` to also catch 401s.
WDYT?

Reply:
Good idea. Only in that place or in both places (after `rateLim` as well?)?